service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/ConversationIdService.java

```diff
@@ -0,0 +1,21 @@
+package uk.nhs.adaptors.gp2gp.common.service;
+
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ConversationIdService {
```

Comment:
I'd consider making a single MDC service with `applyConversationId` and `applyTaskId` methods and a single `resetAll` method. That encapsulates all of the code related to logging diagnostic identifiers in a single place.

I'd question whether we need getters at all. They were used in NHAIS but I'm not sure that was a great design choice.